byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,195 @@
+      destination: end,
+      travelMode: 'DRIVING'
+  };
+  directionsService.route(request, function(response, status) {
+    if (status == 'OK') {
+      directionsRenderer.setDirections(response);
+      findRegions(response);
+      loadRecommendations();
+    } else {
+      alert("Could not calculate route due to: " + status);
+    }
+  });
+}
+
+/**
+ * Goes through steps along route to find the average location
+ * between a step's start and end location. Store these points
+ * in the regions Array.
+ * @param {DirectionsResult directionResult} contains directions
+ * for the route made.
+ */
+function findRegions(directionResult) {
+  const myRoute = directionResult.routes[0].legs[0];
+  for(step of myRoute.steps) {
+    let avgLat = (step.start_location.lat() + step.end_location.lat()) / 2;
+    let avgLng = (step.start_location.lng() + step.end_location.lng()) / 2;
+    let avgLoc = {lat: avgLat, lng: avgLng};
+    if(step.distance.value > MIN_DISTANCE) {
+      regions.push(avgLoc);
+    }
+  }
+  console.log("size of regions: " + regions.length);
+}
+
+/**
+ * Set a timeout to delay the browser.
+ * @param {Number delayMs} number of milliseconds
+ */
+function delayPromise(delayMs) {
+  return new Promise(resolve => setTimeout(resolve, delayMs));
+}
+
+/**
+ * Go through a user's interests and search for places
+ * fitting those interests. Search around the regions
+ * previously found using textSearch.
+ */
+async function loadRecommendations() {
+  for(interest of interests) {
+    for(region of regions) {
+      await delayPromise(250);
+      let request = {
+        location: region,
+        radius: RADIUS,
+        query: interest
+      }
+      const results = await new Promise(resolve => {
+        placesService.textSearch(request, (result, status) => {
+          if(status == "OK") {
+            resolve(result);
+          } else if(status == "OVER_QUERY_LIMIT") {
+            alert("Showing limited results");
+          } else {
+            alert("Status: " + status);
+          }
+        });
+      });
+      addRecommendations(results);
+    }
+  }
+}
+
+/**
+ * Places markers on the locations found from textSearch.
+ * Temporarily limit the amount of suggestions.
+ * TODO: Store more results and limit on UI with option to "show more"
+ * @param {PlaceResults[] results} places found with PlaceResult type.
+ */
+function addRecommendations(results) {
+  let maxRecommendations = 1;
+  let numRecommendations = 0;
+  for (result of results) {
+    console.log(result.name);
+    placeMarker(result);
+    numRecommendations++;
+    if(numRecommendations == maxRecommendations) {
+      break;
+    }
+  }
+}
+
+/**
+ * Places a marker onto the map at the specified location.
+ * @param {PlaceResult place} Object containing information
+ * about a place
+ */
+function placeMarker(place) {
+  let infoWindow = new google.maps.InfoWindow({
+    content: place.name
+  });
+  const image = {
+    url: './images/stopsMarker.png',
```

Comment:
Use an absolute path 'images/...' instead of a relative one.

Reply:
Fixed, included root directory `/images/...`